### ISSUE TITLE
MD003/header-style: bug, setext followed by atx flagged as error with setext_with_atx setting.

### ISSUE THREAD

**Issue by fbartho** [DavidAnson/markdownlint#94]
If you have a markdown file as follows:

```markdown
Title
=====

## Heading

Some stuff
```

And a config of:
```json
  "markdownlint": {
    "header-style": { "style": "setext_with_atx" },
  }
```

You get the following error
```
file: 'file:///Users/fbartho/Code/MD003ErrorExample.md'
severity: 'Warning'
message: 'MD003/header-style: Header style [Expected: setext; Actual: atx]'
at: '4,1'
source: 'markdownlint'
```

It appears to be mis-parsing `setext_with_atx` as simply `setext`

**Comment by DavidAnson** [comment]
Per the documentation, `setext_with_atx` uses `setext` style for header levels 1 and 2 and `atx` style for header levels 3 and above. Therefore, the message above seems correct.

https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md003

**Comment by fbartho** [comment]
Ahh! I read the option name directly, and didn't see in the doc-page that it was specifically for levels 3 & above. I thought it was for any mix & match. Thanks!

Feel free to close this ticket, or turn it into a documentation improvement request? My question is resolved.

**Comment by DavidAnson** [comment]
Because this is the first time this question has come up, I will close it for now. If someone else asks, I will make an effort to improve the documentation. Thanks for the quick reply!
